index.js

```diff
@@ -4,33 +4,71 @@
  * http://mozilla.org/MPL/2.0/.
  */
 
-// const system = require('sdk/system');
+const system = require('sdk/system');
 const pageMod = require('sdk/page-mod');
+const { getActiveView } = require('sdk/view/core');
 const getYouTubeUrl = require('./lib/get-youtube-url.js');
 const getVimeoUrl = require('./lib/get-vimeo-url.js');
 const launchVideo = require('./lib/launch-video');
 const sendMetricsData = require('./lib/send-metrics-data.js');
 const initContextMenuHandlers = require('./lib/context-menu-handlers.js');
-// const makePanelDraggable = require('./lib/make-panel-draggable.js');
-const { getActiveView } = require('sdk/view/core');
+const makePanelDraggable = require('./lib/make-panel-draggable.js');
+const { setTimeout } = require('sdk/timers');
+
+const dimensions = {
+  width: 320,
+  height: 180,
+  minimizedHeight: 40
+};
 
 const panel = require('sdk/panel').Panel({
   contentURL: './default.html',
   contentScriptFile: './controls.js',
-  width: 320,
-  height: 180,
+  width: dimensions.width,
+  height: dimensions.height,
   position: {
     bottom: 10,
     left: 10
   }
 });
 
+// Init the panel's location state
+panel.coords = { bottomOffset: -10, leftOffset: 10 };
+
 getActiveView(panel).setAttribute('noautohide', true);
 
-// check for linux here
-// if (system.platform !== 'linux') {
-//   makePanelDraggable(panel);
-// }
+// Draggability seems to work for windows and mac, but not linux.
+if (system.platform === 'winnt' || system.platform === 'darwin') {
+  // We have to wait until XBL has initialized before changing the 'type'.
+  setTimeout(() => { makePanelDraggable(panel, dimensions); }, 5000);
+}
+
+function adjustHeight(newHeight) {
+  // Resize the panel by changing the height of the panel, iframe, and stack
+  // elements.
+  // On linux, makePanelDraggable isn't invoked, so the stack doesn't exist.
+  const xulPanel = getActiveView(panel);
+  const stack = xulPanel.getElementsByTagName('stack') && xulPanel.getElementsByTagName('stack')[0];
+  const frame = xulPanel.getElementsByTagName('iframe')[0];
```

Comment:
Will there always be an `<iframe>`?
I see on L51, we do a check for `stack` before trying to access `stack[0]`, but here we just directly access `iframe[0]`